flake.nix

```diff
@@ -0,0 +1,71 @@
+{
+  description = "An example Golang package";
+
+  # Nixpkgs / NixOS version to use.
+  inputs.nixpkgs.url = "nixpkgs/nixos-21.11";
+
+  outputs = { self, nixpkgs }:
+    let
+
+      # to work with older version of flakes
+      lastModifiedDate = self.lastModifiedDate or self.lastModified or "19700101";
+
+      # Generate a user-friendly version number.
+      version = builtins.substring 0 8 lastModifiedDate;
+
+      # System types to support.
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
+
+      # Helper function to generate an attrset '{ x86_64-linux = f "x86_64-linux"; ... }'.
+      forAllSystems = nixpkgs.lib.genAttrs supportedSystems;
+
+      # Nixpkgs instantiated for supported system types.
+      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; });
+
+    in
+    {
+
+      # Provide some binary packages for selected system types.
+      packages = forAllSystems (system:
+        let
+          pkgs = nixpkgsFor.${system};
+        in
+        {
+          example = pkgs.buildGoModule {
+            pname = "example";
+            inherit version;
+            # In 'nix develop', we don't need a copy of the source tree
+            # in the Nix store.
+            src = ./src/example;
+
+            # This hash locks the dependencies of this package. It is
+            # necessary because of how Go requires network access to resolve
+            # VCS.  See https://www.tweag.io/blog/2021-03-04-gomod2nix/ for
+            # details. Normally one can build with a fake sha256 and rely on native Go
+            # mechanisms to tell you what the hash should be or determine what
+            # it should be "out-of-band" with other tooling (eg. gomod2nix).
+            # To begin with it is recommended to set this, but one must
+            # remeber to bump this hash when your dependencies change.
```

Comment:
This is Nix specific and independent of the example package really. Also, it's mostly fromnscaffolding. Do you still want me to update the comment?

Reply:
Since it's generated text, no need to update the comment here, but I do think the README should include something about how to bump the hash if the dependencies change.

Reply:
This was fixed in 1dc16f21dc82b4b0fd1b96f6e693db02c2df4055

Reply:
Sent this directly  to Dave but  for posterity I will update the document to mention `nix flake update` as well.  Also, when the flake is  rebuilt it’ll complain that the hashes don’t match. Something like: expected dd54696 but received ee759458 in which case you would copy the expected and replace it with what’s currently in the file.

Reply:
Looks like this README update was done in https://github.com/cisagov/skeleton-golang-package/pull/2/commits/bcfc41ca2d45595ed90cbcc594b4d4027486c786. 👍